core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -437,12 +433,7 @@ private void handleKeycode(int keycode, int source) {
                 break;
             case KeyEvent.KEYCODE_MEDIA_PAUSE:
                 if (status == PlayerStatus.PLAYING) {
-                    mediaPlayer.pause(false, true);
-                }
-                if (UserPreferences.isPersistNotify()) {
-                    mediaPlayer.pause(false, true);
-                } else {
-                    mediaPlayer.pause(true, true);
+                    mediaPlayer.pause(() -> !UserPreferences.isPersistNotify(), false);
```

Comment:
It's fine to force push to the pr.

On Sun, Apr 24, 2016, 9:58 PM domingos86 <notifications@github.com> wrote:

> In
> core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java
> <https://github.com/AntennaPod/AntennaPod/pull/1879#discussion_r60857353>:
>
> > @@ -437,12 +433,7 @@ private void handleKeycode(int keycode, int source) {
> >                  break;
> >              case KeyEvent.KEYCODE_MEDIA_PAUSE:
> >                  if (status == PlayerStatus.PLAYING) {
> > -                    mediaPlayer.pause(false, true);
> > -                }
> > -                if (UserPreferences.isPersistNotify()) {
> > -                    mediaPlayer.pause(false, true);
> > -                } else {
> > -                    mediaPlayer.pause(true, true);
> > +                    mediaPlayer.pause(() -> !UserPreferences.isPersistNotify(), false);
>
> I believe I understand now. First, the fact that it had two calls was
> simply a mistake, as far as I know (based on the commit that introduced it).
> Then, the reinit only applies when streaming and using an intent (media
> buttons, notification, etc), so it means the AP UI isn't open and, running
> on background, we probably want to save some bandwidth.
> It's fixed now.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/AntennaPod/AntennaPod/pull/1879/files/20a9e5908a3ec0021429f34f31489079fc807fd5..24254ca35aca78b9804b58f617975282dd4a4c8a#r60857353>
>